app/src/main/java/com/portfolio/tasky/networking/RequestInterceptor.kt

```diff
@@ -1,19 +1,18 @@
 package com.portfolio.tasky.networking
 
-import com.portfolio.tasky.BuildConfig
+import com.portfolio.tasky.entry.usecases.domain.UserPreferences
 import okhttp3.Interceptor
 import okhttp3.Response
 import java.io.IOException
 import javax.inject.Inject
 
 
-class RequestInterceptor @Inject constructor() :
-    Interceptor {
+class RequestInterceptor @Inject constructor(private val userPreferences: UserPreferences) : Interceptor {
     @Throws(IOException::class)
     override fun intercept(chain: Interceptor.Chain): Response {
         val original = chain.request()
         val requestBuilder = original.newBuilder()
-            .header("x-api-key", BuildConfig.api_key)
```

Comment:
How comes you removed this?

Reply:
Called it directly in the Api interface.

Reply:
Here you'd only need to add this single line to make sure all your requests contain the key. In your API interface you'll need to add this for every single route